### DISCUSSION TITLE
Lab 2 Demo

### DISCUSSION THREAD

**Discussion by ryann-li** [NUS-CG3207/labs#16] [Lab 2] [answered]
Hi, I am confused about the demo requirements. We are able to verify that our processor works by simulating and seeing that the signals are correct (eg/. the WD shows the correct value after doing an add, PC increments properly by 4 each time, things like this). However, is showing this simulated waveform in the demo enough? How do we actual prove this using the board aswell?

If a student from the Monday session or an instructor can help clarify this I would appreciate it a lot. Thank you.

**Comment by kxing28** [comment, marked as answer]
In our group, we made an assembly program that ends up using all the instructions. We put it into Wrapper.v and created a testbench to simulate the program. We also ran the program on the board.

I think the idea of the demo is that if the board was able to run the program normally then the rest of the system is working properly.

We also looked at the simulation and demonstrated that when a specific instruction came in, certain flags were raised.
